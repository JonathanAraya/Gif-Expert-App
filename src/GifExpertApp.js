import { useState } from 'react'
import { AddCategory } from './components/AddCategory'

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['1', '2', '3'])

    // const handleAdd = () => {
    //     setCategories([...categories, '4']);
    // }

    return(
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories } />
        <hr />
        <ol>
            {
                categories.map(( cat ) => {
                    return <li key={ cat }>{cat}</li>
                })
            }
        </ol>
        </>
    )
}

export default GifExpertApp;