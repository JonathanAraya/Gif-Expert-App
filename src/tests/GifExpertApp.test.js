import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas a componente <GifExpertApp />', () => {
    
    test('Debe hacer match con el snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrar una lista de categorias', () => {
        
        const categories = ['Minecraft', 'DOOM']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
