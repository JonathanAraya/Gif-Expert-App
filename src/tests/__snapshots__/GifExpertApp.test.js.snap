// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas a componente <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Minecraft"
      key="Minecraft"
    />
    <GifGrid
      category="DOOM"
      key="DOOM"
    />
  </ol>
</Fragment>
`;
